### PR TITLE
Fix getting http response body in proxytest

### PR DESCRIPTION
**Pull request by anuraaga**

### CODE REVIEW

File: proxywasm/proxytest/http.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func (h *httpHostEmulator) httpHostEmulatorProxyGetBufferBytes(bt internal.Buffe
 	case internal.BufferTypeHttpRequestBody:
 		buf = stream.requestBody
 	case internal.BufferTypeHttpResponseBody:
-		buf = stream.requestBody
+		buf = stream.responseBody
```
Comment (anuraaga) [inline proxywasm/proxytest/http.go:61]: @mathetake This is the main fix

**Comment by anuraaga** [comment]
Hmm maybe I shouldn't have updated the existing test since it'll be trickier updating the e2e on mac, but if you don't have any hints I'll see what I can do

**Comment by mathetake** [comment]
You have to remove the original `content-length` header as at the point of OnHttpRespnoseBody, HTTP headers  already arrived at the client https://github.com/tetratelabs/proxy-wasm-go-sdk/blob/0a6c0043e71dc06b2b85b93fa5d033c6982d0749/examples/http_body/main.go#L79-L89

File: proxywasm/proxytest/http.go (hunk 1 above)
Comment (mathetake) [reply proxywasm/proxytest/http.go:61]: cool

**Comment by mathetake** [comment]
```
2022/08/25 06:34:43 Unsolicited response received on idle HTTP channel starting with "ded body][original body]"; err=<nil>
2022/08/25 06:34:44 Unsolicited response received on idle HTTP channel starting with "ded body][original body]"; err=<nil>
2022/08/25 06:34:44 Unsolicited response received on idle HTTP channel starting with "ded body][original body]"; err=<nil>
2022/08/[25](https://github.com/tetratelabs/proxy-wasm-go-sdk/runs/8009973129?check_suite_focus=true#step:8:26) 06:34:45 Unsolicited response received on idle HTTP channel starting with "ded body][original body]"; err=<nil>
2022/08/25 06:34:45 Unsolicited response received on idle HTTP channel starting with "ded body][original body]"; err=<nil>
2022/08/25 06:34:46 Unsolicited response received on idle HTTP channel starting with "ded body][original body]"; err=<nil>
2022/08/25 06:34:46 Unsolicited response received on idle HTTP channel starting with "ded body][original body]"; err=<nil>
2022/08/25 06:34:47 Unsolicited response received on idle HTTP channel starting with "ded body][original body]"; err=<nil>
2022/08/25 06:34:47 Unsolicited response received on idle HTTP channel starting with "ded body][original body]"; err=<nil>
```

This says that HTTP body is larger than expected (content-length header)

**Comment by anuraaga** [comment]
Thanks a lot @mathetake I was not on the right track at all!

File: e2e/e2e_test.go [hunk 2]
```diff
@@ -95,8 +95,8 @@ func Test_http_body(t *testing.T) {
 	for _, tc := range []struct {
 		op, expBody string
 	}{
-		{op: "append", expBody: `[original body][this is appended body]`},
-		{op: "prepend", expBody: `[this is prepended body][original body]`},
+		{op: "append", expBody: `[original body][this is appended body][this is appended body]`},
```
Comment (mathetake) [inline e2e/e2e_test.go:98]: could we replace/append/prepend only at either Response or Request (let's control the behavior with other header), then there won't be no ambiguity in the response body which part of Plugin create what?

Comment (mathetake) [reply e2e/e2e_test.go:98]: like `curl -XPUT localhost:18000 --data '[initial body]' -H "buffer-operation: prepend" -H "buffer-replace-at: response"` to make replacement happen at response where `buffer-replace-at: response` controls the behavior.


Also, please make change to  https://github.com/tetratelabs/proxy-wasm-go-sdk/blob/main/examples/http_body/README.md

Comment (anuraaga) [reply e2e/e2e_test.go]: Sure updated

File: examples/http_body/README.md [hunk 3]
```diff
@@ -13,3 +13,15 @@ $ curl -XPUT localhost:18000 --data '[initial body]' -H "buffer-operation: appen
 $ curl -XPUT localhost:18000 --data '[initial body]' -H "buffer-operation: replace"
 [this is replaced body]
 ```
+
+To modify the response:
+```
+****$ curl -XPUT localhost:18000 --data '[initial body]' -H "buffer-operation: prepend" -H "buffer-replace-at: response"
```
Comment (mathetake) [inline examples/http_body/README.md:19]: Garbage 🕵️ 

```suggestion
$ curl -XPUT localhost:18000 --data '[initial body]' -H "buffer-operation: prepend" -H "buffer-replace-at: response"
```

File: e2e/e2e_test.go [hunk 4]
```diff
@@ -92,34 +92,43 @@ func Test_http_body(t *testing.T) {
...
-				defer res.Body.Close()
-				body, err := io.ReadAll(res.Body)
-				require.NoError(t, err)
-				return string(body) == tc.expBody &&
-					checkMessage(stdErr.String(), []string{
-						`original request body: [original body]`},
-						[]string{"failed to"},
-					) && checkMessage(string(body), []string{tc.expBody}, nil)
-			}, 5*time.Second, 500*time.Millisecond, stdErr.String())
+		t.Run(mode, func(t *testing.T) {
```
Comment (mathetake) [inline e2e/e2e_test.go:99]:
```suggestion
		mode := mode
		t.Run(mode, func(t *testing.T) {
```

File: e2e/e2e_test.go [hunk 5]
```diff
@@ -92,34 +92,43 @@ func Test_http_body(t *testing.T) {
...
+			for _, tc := range []struct {
+				op, expBody string
+			}{
+				{op: "append", expBody: `[original body][this is appended body]`},
+				{op: "prepend", expBody: `[this is prepended body][original body]`},
+				{op: "replace", expBody: `[this is replaced body]`},
+				// Should fall back to to the replace.
+				{op: "invalid", expBody: `[this is replaced body]`},
+			} {
+				t.Run(tc.op, func(t *testing.T) {
```
Comment (mathetake) [inline e2e/e2e_test.go:109]: while you are here :bow:
```suggestion
				tc := tc
				t.Run(tc.op, func(t *testing.T) {				
```

File: e2e/e2e_test.go (hunk 4 above)
Comment (anuraaga) [reply e2e/e2e_test.go:99]: `mode` is just a string so don't think this one is actually needed

**Review by mathetake — APPROVED** [review]

**Comment by mathetake** [comment]
https://github.com/tetratelabs/proxy-wasm-go-sdk/blob/494d32c81bfb2f3f63c0797f6f6abba4a78e0bfa/.github/workflows/workflow.yaml#L122 another place to fix

**Comment by anuraaga** [comment]
Ouch - in a future PR will try swapping cache for [artifacts](https://github.com/actions/upload-artifact) as I think those are job-scoped so wouldn't have the chance of loading ancient examples that pass e2es with failing code 🙀

**Comment by mathetake** [comment]
👍

**Comment by mathetake** [comment]
started failing :D

**Comment by mathetake** [comment]
oh 0.25 won't work on envoy...?

**Comment by anuraaga** [comment]
@mathetake Hmm going to need to investigate - in coraza-wasm-filter I'm using 0.25.0 with envoy 1.22 in docker and it does seem to work

**Comment by mathetake** [comment]
ok

**Pull request merged**